platforms/core-execution/persistent-cache/src/main/java/org/gradle/cache/internal/locklistener/UnixDomainSocketFileLockCommunicator.java

```diff
@@ -0,0 +1,216 @@
+        byte[] bytes = new byte[bytesRead];
+        buffer.flip();
+        buffer.get(bytes);
+        return bytes;
+    }
+
+    @Override
+    public FileLockPacketPayload decode(FileLockPacket receivedPacket) {
+        try {
+            return FileLockPacketPayload.decode(receivedPacket.getData(), receivedPacket.getLength());
+        } catch (IOException e) {
+            if (!stopped) {
+                throw new RuntimeException(e);
+            }
+            throw new GracefullyStoppedException();
+        }
+    }
+
+    @Override
+    public void confirmUnlockRequest(SocketAddress address, long lockId) {
+        try {
+            byte[] bytes = FileLockPacketPayload.encode(lockId, UNLOCK_REQUEST_CONFIRMATION);
+            send(address, bytes);
+        } catch (IOException e) {
+            if (!stopped) {
+                throw new RuntimeException(e);
+            }
+            throw new GracefullyStoppedException();
+        }
+    }
+
+    @Override
+    public void confirmLockRelease(Set<SocketAddress> addresses, long lockId) {
+        byte[] packet = FileLockPacketPayload.encode(lockId, LOCK_RELEASE_CONFIRMATION);
+        for (SocketAddress address : addresses) {
+            try {
+                LOGGER.debug("Confirming lock release to Gradle process at address {} for lock with id {}.", address, lockId);
+                send(address, packet);
+            } catch (IOException e) {
+                if (!stopped) {
+                    LOGGER.debug("Failed to confirm lock release to Gradle process at address {} for lock with id {}.", address, lockId);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void stop() {
+        try {
+            stopped = true;
+            serverChannel.close();
+            Files.deleteIfExists(unixDomainSocketPath(thisProcessAddress));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public int getPort() {
+        // This is a Unix domain socket, so there is no port, we use pid instead
+        // TODO: Use long instead of int for port
+        return stopped ? -1 : (int) pid;
+    }
+
+    private static ServerSocketChannel openAndBindServerSocketChannel(SocketAddress thisProcessAddress) {
+        try {
+            System.out.println("Opening server socket channel at address: " + thisProcessAddress);
+            ServerSocketChannel serverSocketChannel = UnixDomainSocketUtil.openUnixServerSocketChannel();
+            return serverSocketChannel.bind(thisProcessAddress);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @VisibleForTesting
+    SocketAddress unixDomainSocketAddressOf(int ownerPort) {
+        // TODO: Where to put domain sockets? If we put them in gradleUserHomeDir, we can get
+        //  java.net.SocketException: Unix domain path too long
+        File unixDomainSocketsDir = new File("/tmp/gradle");
```

Comment:
I'm unsure where macOS expects them, but the Linux standard is to put these things in `$XDG_RUNTIME_DIR` or failing that `$TMPDIR`.

Looking at https://github.com/dirs-dev/directories-jvm/blob/main/src/main/java/dev/dirs/ProjectDirectories.java#L237, it seems like there's no equivalent directory for macOS/Windows.

So I think using `$XDG_RUNTIME_DIR/gradle` on Linux if possible, and using the actual temporary directory via `Files.createTempDirectory("gradle")` otherwise, would be the best option.

Reply:
Thanks, it might make more sense to use that. 
Note that I checked `System.getProperty("java.io.tmpdir")` value and it looks like it can be long path too, e.g. `/Users/asodja/workspace/gradle/subprojects/core/build/tmp` for some tests when running locally.

So maybe we need a differet fallback. I wonder what others use, e.g. I know Docker uses `docker.sock`